spec/integration/cleanup_spec.rb

```diff
@@ -132,5 +132,12 @@
         assert all_triggers.find { |t| t =~ /lhmt_(.*)_users/}
       end
     end
+
+    describe 'failed_name' do
+      it 'should provide a table name with a maximum length of 64' do
+        cleanup = Lhm::Cleanup::Current.new(nil, 'some_excessively_long_table_name_oh_dear_code_who_would_even_do_this_and_why', nil)
+        assert_equal cleanup.instance_eval { failed_name }.size, 64
```

Comment:
`instance_eval` this is an anti-pattern which we should not tolerate. If the code is not easily testable, the code needs to change.

Reply:
Can you explain why instance_eval a bad idea?  Or at least give me a hint why so I can google up an article to help me understand.

Reply:
`instance_eval` is a way to "reach inside" an instance of a class and do whatever you wish. You can manipulate the behavior of the instance or query it's internals. It's a powerful, but dangerous tool. In this particular context, we're using it to make assertions about the internals of cleanup. This suggests that what we want to test isn't publicly available which suggests that the code responsible for this needs to be extracted into it's own class, which can be tested via normal means. Happy to dig into this further if you have other questions. Thanks for asking!

Reply:
👍 thanks for the detail.